docs/read-more/how-to-write-a-sequence.md

```diff
@@ -0,0 +1,221 @@
+
+Sequence is a program that produces, consumes or transforms data. It’s a function or an array of functions. They typically look somewhat like this:
+
+```ts
+/** The function parameters are
+* input stream
+* ...params passed to instance on start
+*/
+export default function(input, param1, param2) {
+    const out = new PassThrough()
+
+    input.on('data', (data) => {
+        out.write('Hello ' + data.toString())
+    })
+
+    input.on('end', () => {
+        this.logger.debug('Input stream ended')
+    })
+
+    // it returns an output stream
+    return out
+}
+```
+
+## Producing data (output stream)
+To stream data from a sequence we need it to return values over time. Some constructs in JavaScript that enable that are NodeJS streams, Generators and Iterables. Whatever you return from your sequence will be your ***output*** stream. We can choose whatever the solution is right for us. Here are some examples of a sequence producing a stream of integers every second.  
+Using a stream:
+```ts
+export default function() {
+    const out = new PassThrough()
+
+    let i = 0
+    setInterval(() => {
+        out.write(i++)
```

Comment:
We need to add drain logic here. Otherwise our customers will write programs that exhaust all the memory rather quickly

Reply:
fixed